lib/publish.ts

```diff
@@ -129,10 +142,12 @@ function doPublish(packageDir: string, gitRemoteUrl: string, params: Params): Pr
 
     function packPackageIntoTarball() {
         return directoryReady
-            .then(() => exec(`npm pack "${packageDir}"`, { cwd: packDir }))
+            .then(() => run([`npm`, `pack`, `${packageDir}`], { cwd: packDir }))
             .then(() => {
                 // pack succeeded! Schedule a cleanup and return the full path
-                cleanupOperations.push(exec(`npm cache clean ${params.originalPackageInfo.name}@${params.originalPackageInfo.version}`));
+                cleanupOperations.push(
+                  parseOut(run([`npm`, `cache`, `clean`, `${params.originalPackageInfo.name}@${params.originalPackageInfo.version}`]))
```

Comment:
@Mik317 , this line raises an exception saying `npm cache clear` doesn't accept any arguments. If the goal here is to verify that the cache is consistent, then we could use `npm cache verify` as it's the one recommended by `npm` itself. I'd love to hear your thoughts on this, or I guess we could leave it to the maintainers hands, I suppose.